app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/TrackerBlockingVpnService.kt

```diff
@@ -236,15 +244,17 @@ class TrackerBlockingVpnService : VpnService(), CoroutineScope by MainScope() {
         }
     }
 
-    private suspend fun establishVpnInterface() {
+    private suspend fun createTunnelInterface() {
         tunInterface = Builder().run {
-            vpnNetworkStack.addresses().forEach { addAddress(it.key, it.value) }
+            val config = vpnNetworkStack.onPrepareVpn().getOrThrow()
```

Comment:
nit, I would call this method `startVpn` and pass the resulting config as parameter here.

Reply:
Fixed!